test/Models/CesiumSpec.ts

```diff
@@ -296,6 +296,120 @@ describeIfSupported("Cesium Model", function () {
       );
     });
   });
+
+  describe("Cesium Terrain Extra Tests", function () {
+    // declare different variables here, we need new instances to test
+    let terria2: Terria;
+    let terriaViewer2: TerriaViewer;
+    let container2: HTMLElement;
+    let cesium2: Cesium;
+
+    beforeEach(function () {
+      terria2 = new Terria({
+        baseUrl: "./"
+      });
+      terriaViewer2 = new TerriaViewer(
+        terria2,
+        computed(() => [])
+      );
+      container2 = document.createElement("div");
+      container2.id = "container2";
+      document.body.appendChild(container2);
+    });
+
+    afterEach(function () {
+      cesium2?.destroy();
+      document.body.removeChild(container2);
+    });
+
+    it("should thow an warning when cesiumIonAccessToken is invalid", async function () {
+      runInAction(() => {
+        terria2.workbench.removeAll();
+        // Set an invalid token for the test
+        terria2.configParameters.cesiumIonAccessToken = "expired_token";
+      });
+      // Instantiate Cesium object with the invalid token
+      cesium2 = new Cesium(terriaViewer2, container2);
+
+      await cesium2.terrainProvider.readyPromise.catch(() => {});
+      // Wait a few ticks to allow for delay in adding event listener to terrainProvider in Cesium.ts
+      await runLater(() => {}, 5);
```

Comment:
Can we get rid of this if we pass `fireImmediately: true` as an option to the [autorun](https://github.com/TerriaJS/terriajs/blob/8a051a7845c6625c53bc50e76920060ccdf22d35/lib/Models/Cesium.ts#L391-L402) call? 

It is better not to use delays in test where we can to avoid slowing down tests.

Reply:
The problem is that the token error handler is added to the terrain provider after `readyPromise` resolves, but somehow awaiting the readyPromise in this test resolves first. The proper fix would be to expose a promise within the Cesium class for the terrain provider ([fromPromise](https://github.com/mobxjs/mobx-utils/tree/5.6.2#frompromise) could be useful for this).

Reply:
The `fireImmediately: true` option doesnt work in this case

Reply:
I am making a separate ticket for "The proper fix would be to expose a promise within the Cesium class for the terrain provider ([fromPromise](https://github.com/mobxjs/mobx-utils/tree/5.6.2#frompromise) could be useful for this)."

Reply:
Here: https://github.com/TerriaJS/terriajs/issues/6690